openeo/extra/job_management.py

```diff
@@ -325,33 +346,28 @@ def run_jobs(
             # Resume from existing db
             _log.info(f"Resuming `run_jobs` from existing {job_db}")
             df = job_db.read()
```

Comment:
This overwrites the df argument of the function, causing a lot of confusion to new job manager users. I would suggest adding a new argument to the function (something like `overlap_resolver` or `df_mode`, although I am not final at all on this name) which allows the user to state that if a job_db already exists the df assed can either be ignored (`ignore`), new lines added to the job_db (`append`) or all lines in the df should be overwritten in de job_db (`overwrite`).

This allows a user to add new jobs to an existing job_db, allows a user to overwrite for example failed jobs (perhaps with new job arguments like more memory), and the default value could be the ignore option to remain backwards compatible.

Lastly I would make the df argument optional with an exception being thrown when there is no job_db and no df, as currently the df is simply not used when a job_db is present but still required.

Reply:
valid comment, but note that it applies to existing behaviour.
I have some other ideas to further refactoring job manager, but may need to keep this PR small to get it through review.